File: imagepicker/src/main/java/com/esafirm/imagepicker/helper/IpLogger.java
package com.esafirm.imagepicker.helper;

import android.util.Log;

public class IpLogger {

    private static final String TAG = "ImagePicker";

    private static IpLogger INSTANCE;

    private boolean isEnable = true;

    public static IpLogger getInstance() {
        if (INSTANCE == null) {
            INSTANCE = new IpLogger();
        }
        return INSTANCE;
    }

    private IpLogger() {
    }

    public void setEnable(boolean enable) {
        isEnable = enable;
    }

    public void d(String message) {
        if (isEnable) {
            Log.d(TAG, message);
        }
    }

    public void e(String message) {
        if (isEnable) {
            Log.e(TAG, message);
        }
    }

    public void w(String message) {
        if (isEnable) {
            Log.w(TAG, message);
        }
    }
}

File: imagepicker/src/main/java/com/esafirm/imagepicker/helper/IpLogger.kt
package com.esafirm.imagepicker.helper

import android.util.Log

object IpLogger {

    private const val TAG = "ImagePicker"

    private var isEnable = true

    fun setEnable(enable: Boolean) {
        isEnable = enable
    }

    fun d(message: String?) {
        if (isEnable && message != null) {
            Log.d(TAG, message)
        }
    }

    fun e(message: String?) {
        if (isEnable && message != null) {
            Log.e(TAG, message)
        }
    }

    fun w(message: String?) {
        if (isEnable && message != null) {
            Log.w(TAG, message)
        }
    }
}